File: 프로그래머스/2/64065. 튜플/튜플.js
function solution(s) {
    var answer = [];
    
    s = s.split("},{")
    const n = s.length
    s[0] = s[0].replace(/^\{+|\}+$/g, '')
    s[n-1] = s[n-1].replace(/^\{+|\}+$/g, '')
    
    s.forEach((x, i) => {
        s[i] = x.split(",")
    })
    
    s.sort((a, b) => {
        if (a.length > b.length) {
            return 1
        } else {
            return -1
        }
    })
    
    const cnt = {}
    s[n-1].forEach(x => {
        if (x in cnt) {
            cnt[x] += 1
        } else {
            cnt[x] = 1
        }
    })

    for (let a = 0; a < n; a++) {
        for (let b = 0; b< s[a].length; b++) {
            if (cnt[s[a][b]]) {
                cnt[s[a][b]] -= 1
                answer.push(parseInt(s[a][b]))
            }        
        }
    }
    return answer;
}

File: 프로그래머스/2/64065. 튜플/튜플.py
from collections import defaultdict
def solution(s):
    answer = []
    s = s.split("},{")
    N = len(s)
    
    s[0] = s[0].lstrip("{")
    s[N-1] = s[N-1].rstrip("}")
    
    for idx in range(len(s)):
        s[idx] = s[idx].split(",")
        
    s.sort(key= lambda x: len(x))
    cnt = defaultdict(int)
    
    for i in s[N-1]:
        cnt[i] += 1
        
    for n in range(N):
        for m in range(len(s[n])):
            if cnt[s[n][m]]:
                answer.append(int(s[n][m]))
                cnt[s[n][m]] -= 1
                
    return answer